File: app/src/main/java/com/hault/codex_java/data/model/World.java
package com.hault.codex_java.data.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "worlds")
public class World {
    @PrimaryKey(autoGenerate = true)
    public int id;
    public String name;
    public String description;

    public World(String name, String description) {
        this.name = name;
        this.description = description;
    }
}

File: app/src/main/java/com/hault/codex/data/model/World.kt
package com.hault.codex.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "worlds")
data class World(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val name: String,
    val description: String?
)
